### DISCUSSION TITLE
NSI brands/shop/alcohol with no obvious scraper

### DISCUSSION THREAD

**Discussion by CloCkWeRX** [alltheplaces/alltheplaces#6918] [Ideas]
- [x] Check all for a store finder page
- [x] Check for existing spiders

----

Fetched 102 brands/shop/alcohol from NSI
Missing by wikidata: 69

"1. day", "Q108149927" 
https://www.1day.sk/ (Cloudflare)
       -> https://www.wikidata.org/wiki/Q108149927
       -> https://www.wikidata.org/wiki/Special:EntityData/Q108149927.json

"ABC (North Carolina)", "Q7054415"
https://www.abc.nc.gov/ - alcohol and beverage comission?
       -> https://www.wikidata.org/wiki/Q7054415
       -> https://www.wikidata.org/wiki/Special:EntityData/Q7054415.json

"Alcohol & Tabakoff", "Q111364479"
https://alcoholtabakoff.com/magazini/ - bespoke HTML
       -> https://www.wikidata.org/wiki/Q111364479
       -> https://www.wikidata.org/wiki/Special:EntityData/Q111364479.json

"Alko", "Q1849187"
https://www.alko.fi/en/shops-services - bespoke HTML
       -> https://www.wikidata.org/wiki/Q1849187
       -> https://www.wikidata.org/wiki/Special:EntityData/Q1849187.json

"Boxer Liquors", "Q122766666" 
https://www.boxer.co.za/ (449 stores, no locator)
       -> https://www.wikidata.org/wiki/Q122766666
       -> https://www.wikidata.org/wiki/Special:EntityData/Q122766666.json

"Henry's", "Q111017798"
https://liquorland.co.nz/ (Defunct?)
       -> https://www.wikidata.org/wiki/Q111017798
       -> https://www.wikidata.org/wiki/Special:EntityData/Q111017798.json

"La Cervoiserie", "Q101788539"  
https://www.cervoiserie.com/contact/ - 1 location?
       -> https://www.wikidata.org/wiki/Q101788539
       -> https://www.wikidata.org/wiki/Special:EntityData/Q101788539.json

"Oddbins", "Q7077472"
https://www.oddbins.com/ - Shopify - defunct. https://www.thedrinksbusiness.com/2023/11/oddbins-closes-doors-on-uk-stores/
       -> https://www.wikidata.org/wiki/Q7077472
       -> https://www.wikidata.org/wiki/Special:EntityData/Q7077472.json

"Shoprite LiquorShop", "Q1857639"
https://www.shoprite.co.za/store-finder
       -> https://www.wikidata.org/wiki/Q1857639
       -> https://www.wikidata.org/wiki/Special:EntityData/Q1857639.json

"Spec's", "Q7574415"
https://specsonline.com/locations/ - weird bot protection
       -> https://www.wikidata.org/wiki/Q7574415
       -> https://www.wikidata.org/wiki/Special:EntityData/Q7574415.json

"The Bottle-O", "Q111015122"
https://thebottleo.co.nz/pages/about-us single store
       -> https://www.wikidata.org/wiki/Q111015122
       -> https://www.wikidata.org/wiki/Special:EntityData/Q111015122.json

"Аванти", "Q111364476"
https://avanti-bg.com/shops/
       -> https://www.wikidata.org/wiki/Q111364476
       -> https://www.wikidata.org/wiki/Special:EntityData/Q111364476.json

"Ароматный мир", "Q109852336"
https://amwine.ru/customers/shops/
       -> https://www.wikidata.org/wiki/Q109852336
       -> https://www.wikidata.org/wiki/Special:EntityData/Q109852336.json

"Біле Сухе", "Q121464345"
https://bilesuhe.com.ua/ - no storefinder?
       -> https://www.wikidata.org/wiki/Q121464345
       -> https://www.wikidata.org/wiki/Special:EntityData/Q121464345.json

"Бристоль", "Q59155583"
https://bristol.ru/shops
       -> https://www.wikidata.org/wiki/Q59155583
       -> https://www.wikidata.org/wiki/Special:EntityData/Q59155583.json

"Главпиво", "Q110138058"
https://glav-beer.ru/
https://glav-beer.ru/addresses/spb/
       -> https://www.wikidata.org/wiki/Q110138058
       -> https://www.wikidata.org/wiki/Special:EntityData/Q110138058.json

"Градусы", "Q110137876"
https://gradusi.net/ - VPN blocker
       -> https://www.wikidata.org/wiki/Q110137876
       -> https://www.wikidata.org/wiki/Special:EntityData/Q110137876.json

"Красное&Белое", "Q24933790"
https://krasnoeibeloe.ru/address/
       -> https://www.wikidata.org/wiki/Q24933790
       -> https://www.wikidata.org/wiki/Special:EntityData/Q24933790.json

"Минск Кристалл", "Q6869455"
http://www.kristallvodka.com/ - dead domain? wikidata wrong?
       -> https://www.wikidata.org/wiki/Q6869455
       -> https://www.wikidata.org/wiki/Special:EntityData/Q6869455.json

"Норман", "Q109907481"
https://norman-retail.ru/contacts/stores/index.php
       -> https://www.wikidata.org/wiki/Q109907481
       -> https://www.wikidata.org/wiki/Special:EntityData/Q109907481.json

"Пивточка", "Q110138021"
https://pivtochka.com/main#adresa
       -> https://www.wikidata.org/wiki/Q110138021
       -> https://www.wikidata.org/wiki/Special:EntityData/Q110138021.json

"РосАл", "Q110137999"
https://rosal24.ru/vybrat-magazina
       -> https://www.wikidata.org/wiki/Q110137999
       -> https://www.wikidata.org/wiki/Special:EntityData/Q110137999.json

"Чарка до свята", "Q118120356"
https://www.alcogurt.com/20-ovacia/26-perelikmagov
       -> https://www.wikidata.org/wiki/Q118120356
       -> https://www.wikidata.org/wiki/Special:EntityData/Q118120356.json

"カクヤス", "Q11293990"
https://www.kakuyasu.co.jp/store/app/shop/shop_list/
       -> https://www.wikidata.org/wiki/Q11293990
       -> https://www.wikidata.org/wiki/Special:EntityData/Q11293990.json

"やまや", "Q17224765"
https://www.yamaya.jp/ wrong wikidata
       -> https://www.wikidata.org/wiki/Q17224765
       -> https://www.wikidata.org/wiki/Special:EntityData/Q17224765.json

"リカーマウンテン", "Q11347883"
       -> https://www.wikidata.org/wiki/Q11347883
       -> https://www.wikidata.org/wiki/Special:EntityData/Q11347883.json

**Comment by CloCkWeRX** [comment]
https://github.com/alltheplaces/alltheplaces/issues/6052

**Comment by davidhicks** [comment]
> "Alcohol & Tabakoff", "Q111364479"
> https://alcoholtabakoff.com/magazini/ - bespoke HTML
> -> https://www.wikidata.org/wiki/Q111364479
> -> https://www.wikidata.org/wiki/Special:EntityData/Q111364479.json

No coordinates for this one as it relies upon address lookup via Google Maps API.

**Comment by davidhicks** [comment]
> "Alko", "Q1849187"
> https://www.alko.fi/en/shops-services - bespoke HTML
> -> https://www.wikidata.org/wiki/Q1849187
> -> https://www.wikidata.org/wiki/Special:EntityData/Q1849187.json

This uses Incapsula bot protection :(

**Comment by CloCkWeRX** [comment]
@davidhicks thanks for turning the good prospects into tickets.

If this approach is useful, happy post/run through the other generated ones to investigate - didn't want to spam the project with low quality scraper requests that were impossible to fill, so slowed a bit.

**Reply by davidhicks** [reply]
Thanks @CloCkWeRX 

These lists are an excellent help. I'm happy to keep taking the useful brands/operators out of these lists and triaging them as issues for people (including myself) to work on.

**Reply by CloCkWeRX** [reply]
Okay, well finished off checking for store finders for the last few.

Might think about changing the generation script to output github issue templates

**Reply by davidhicks** [reply]
I've raised a new Wikidata property request at https://www.wikidata.org/wiki/Wikidata:Property_proposal/Generic#location_information_URL

I think it'd be good, if we can, to capture the landing page of store locators/operator facility lists/etc in Wikidata, as this can then be consumed by other projects, particularly those related to OSM such as Name Suggestion Index.

We don't capture these URLs currently in AllThePlaces, but it'd be helpful to have these URLs available, particularly if https://github.com/alltheplaces/alltheplaces/pull/6526 is implemented fully.

**Reply by CloCkWeRX** [reply]
Nice. Surprisingly, schema.org or https://www.heppnetz.de/ontologies/goodrelations/v1.html#Location don't seem to have a dedicated "store directory" concept, relying on just have list/array semantics.

Wonder if there is anything with open search or other web component/site map description ontologies that says "This is a store finder, there is a geolocation search endpoint, these urls describe a gr:Location"

Would make your auto discovery code a bunch simpler!

**Reply by davidhicks** [reply]
The auto discovery work in progress of #6526 needs more work obviously. The idea eventually would be that Wikidata stores store finder landing pages for brands/operators, and we can then use these URLs to automatically detect common store finder APIs. Specifying property [P11707 location URL match pattern](https://www.wikidata.org/wiki/Property:P11707) would also allow us to possibly automate or semi-automate creation of SitemapSpider+StructuredDataSpider or possibly even CrawlSpider+StructuredDataSpider.

**Comment by CloCkWeRX** [comment]
I feel like most of the low hanging fruit has now been dealt with from here.
Japanese/Russian/etc stores I can find with google translate, but it's really hit and miss.

**Comment by peternewman** [comment]
> "The Bottle-O", "Q111015122"
> https://thebottleo.co.nz/pages/about-us single store

There's a full store list here:
https://shop.thebottleo.co.nz/

> -> https://www.wikidata.org/wiki/Q111015122
> -> https://www.wikidata.org/wiki/Special:EntityData/Q111015122.json

I've opened #14891
